src/Datagrid/Pager.php

```diff
@@ -43,22 +47,12 @@ class Pager extends BasePager
      */
     public function computeNbResult()
     {
-        $countQuery = clone $this->getQuery();
-
-        if (\count($this->getParameters()) > 0) {
-            $countQuery->setParameters($this->getParameters());
-        }
-
-        if (\count($this->getCountColumn()) > 1) {
-            $this->countCompositePrimaryKey($countQuery);
-        } else {
-            $this->countSinglePrimaryKey($countQuery);
+        if (!$this->getQuery() instanceof ProxyQuery) {
+            throw new \LogicException('Only '.ProxyQuery::class.' is supported');
         }
```

Comment:
There is now a `Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQueryInterface.

```suggestion
        if (!$this->getQuery() instanceof ProxyQueryInterface) {
            throw new \InvalidValueException(sprintf('The datagrid query MUST implements %s', .ProxyQueryInterface::class.'));
        }
```
@phansys may have a better exception type and exception message.